src/middleware/middleware.ts

```diff
@@ -11,11 +13,19 @@ import { useNavStore } from '~/stores/nav'
  * add `?embedded=false` to the end of the URL.
  *
  * Messages sent to the outer window have the following format:
- * {type: <event type>, value: <event value>}.
+ * `{type: <event type>, value: <event value>}`.
  * Currently, one event type is used:
  * - `urlChange` sends the relative path of the URL on every URL change.
  */
-export default function ({ query, route, $pinia }) {
+export default async function ({
+  query,
+  route,
+  $pinia,
+}: {
+  query: Record<string, string>
+  route: Route
+  $pinia: Pinia
+}): Promise<void> {
```

Comment:
I guess this conflicts with #1228, should we merge that first then update this afterwards to use the standard implementation for the types?